src/java/org/apache/poi/hssf/usermodel/helpers/HSSFColumnShifter.java

```diff
@@ -0,0 +1,102 @@
+package org.apache.poi.hssf.usermodel.helpers; 
+ 
+import org.apache.poi.hssf.usermodel.HSSFCell; 
+import org.apache.poi.hssf.usermodel.HSSFRow; 
+import org.apache.poi.ss.formula.FormulaShifter; 
+import org.apache.poi.ss.usermodel.Cell; 
+import org.apache.poi.ss.usermodel.CellType; 
+import org.apache.poi.ss.usermodel.Row; 
+import org.apache.poi.ss.usermodel.Sheet; 
+import org.apache.poi.ss.usermodel.helpers.ColumnShifter; 
+import org.apache.poi.util.POILogFactory; 
+import org.apache.poi.util.POILogger; 
+import org.apache.poi.xssf.usermodel.helpers.XSSFRowShifter; 
+ 
+public class HSSFColumnShifter extends ColumnShifter{ 
+    private static final POILogger logger = POILogFactory.getLogger(XSSFRowShifter.class); 
+     
+    private int firstShiftColumnIndex;  
+    private int lastShiftColumnIndex;  
+    private int shiftStep; 
+     
+    public HSSFColumnShifter(Sheet sh, FormulaShifter shifter) { 
+        super(sh, shifter); 
+    } 
+ 
+    public void shiftColumns(int firstShiftColumnIndex, int lastShiftColumnIndex, int step){ 
+        this.firstShiftColumnIndex = firstShiftColumnIndex; 
+        this.lastShiftColumnIndex = lastShiftColumnIndex; 
+        this.shiftStep = step; 
+        if(shiftStep > 0) 
+            shiftColumnsRight(); 
+        else if(shiftStep < 0) 
+            shiftColumnsLeft(); 
+//        formulaShiftingManager.updateFormulas(); 
+    } 
+    /** 
+     * Inserts shiftStep empty columns at firstShiftColumnIndex-th position, and shifts rest columns to the right  
+     * (see constructor for parameters) 
+     */ 
+ 
+    private void shiftColumnsRight(){ 
```

Comment:
I prefer it that way because direction of loops and shifting are different. Surely I could introduce some parameters, and get over it, but I think that that single method would be much harder to read and understand.
Anyway, if you really want me to merge them, I'll do it.